### PR TITLE
Add Medinstruct

### PR DESCRIPTION
**Pull request by mmhamdy**
Hi everyone 👋

This PR adds the MedInstruct Dataset (with the downloader, preparer, and mapper). All tests passed successfully.

Tell me what you think, happy to make any changes.

### CODE REVIEW

**Review by shayne-longpre — APPROVED** [review]
Approved to merge after minor comments! Looks great

File: data_summaries/MedInstruct.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            "UC Santa Barbara"
+        ],
+        "Licenses": [
+            {
+                "License": "Apache License 2.0",
+                "License URL": "https://github.com/XZhang97666/AlpaCare/blob/master/LICENSE"
+            }
+        ],
+        "License Notes": "",
+        "License Verified By": "",
```
[the remark this answers is not in the document]
Comment (mmhamdy) [reply data_summaries/MedInstruct.json:45]: Oh, yes! I just thought it was supposed to be the reviewer😅

File: data_summaries/MedInstruct.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        ],
+        "Task Categories": [
+            "BioMedical Question Answering",
+            "Clinical Diagnosis Question Answering",
+            "Instructional Question Answering",
+            "Medical Diagnosis Question Answering",
+            "Medical Question Answering",
+            "Question Answering"
+        ],
+        "Text Sources": [],
```
[the remark this answers is not in the document]
Comment (mmhamdy) [reply data_summaries/MedInstruct.json]: Added text sources

File: data_summaries/MedInstruct.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        ],
+        "Format": [
+            "Zero-shot"
+        ],
+        "Human Annotation": "No",
+        "Derived from Datasets": [],
+        "Creators": [
+            "UC Santa Barbara"
+        ],
+        "Licenses": [
```
[the remark this answers is not in the document]
Comment (mmhamdy) [reply data_summaries/MedInstruct.json:41]: Added OpenAI and removed the Apache-2.0

**Pull request merged**